plugins/cpp/parser/src/clangastvisitor.h

```diff
@@ -141,6 +141,50 @@ class ClangASTVisitor : public clang::RecursiveASTVisitor<ClangASTVisitor>
     return b;
   }
 
+
+  bool	TraverseCXXForRangeStmt (clang::CXXForRangeStmt *forRangeStmt){
+    return Base::TraverseCXXForRangeStmt(forRangeStmt);
+  }
+
+  bool VisitCXXForRangeStmt(clang::CXXForRangeStmt *forRangeStmt)
+  {
+    model::CppAstNodePtr astNode = std::make_shared<model::CppAstNode>();
+    
+    clang::DeclRefExpr* declRefExpr = llvm::dyn_cast<clang::DeclRefExpr>(forRangeStmt->getRangeInit());
+
+    clang::Stmt* stmt = llvm::dyn_cast<clang::Stmt>(forRangeStmt->getRangeInit());
+
+    clang::CallExpr* callExpr = llvm::dyn_cast<clang::CallExpr>(stmt);
```

Comment:
When we have a function or method call in the range.based for loop, the `forRangeStmt->getRangeInit()` will be a `ExprWithCleanups` statement, and one of its descendant in the AST will be the `CallExpr` or `CXXMemberCallExpr`.

Eg. for a function call:
![image](https://user-images.githubusercontent.com/7405458/218608968-10bb8d8e-a0fd-430c-a5ff-d8127c526984.png)

And for a method call:
![image](https://user-images.githubusercontent.com/7405458/218608880-c68069ae-fe4c-4c0e-9b29-27a73cbe74d8.png)

Reply:
@whisperity: is there an easy way to request the first descendant of a node which matches a certain type? E.g. to get the first `CallExpr` descendant of the `ExprWithCleanups` node in the AST?

Reply:
Unfortunately, I think no. Every node is implemented with different fields and accessor methods. The whole `RecursiveASTVisitor` is using lots of generated code in the background to facilitate the traversals. We could try using the _AST Matchers_ library (that Clang-Tidy also uses...) that has a generic `has()` matcher which can bind a child node.